core/contract/kernel/kernel.go

```diff
@@ -495,104 +526,124 @@ func (k *Kernel) validateUpdateBlockChainData(desc *contract.TxDesc) error {
 	return nil
 }
 
+func (k *Kernel) runCreateBlockChain(desc *contract.TxDesc) error {
+	// 需要校验，否则容易panic
+	bcName, bcData, err := k.validateCreateBC(desc)
+	if err != nil {
+		return err
+	}
+	k.log.Debug("contract: create block chain", "from", k.bcName, "toCrate", bcName)
+	if k.bcName != "xuper" {
+		k.log.Warn("only xuper chain can create side-chain", "bcName", k.bcName)
+		return ErrPermissionDenied
+	}
+	if !desc.Tx.FromAddrInList(k.newChainWhiteList) && !k.disableCreateChainWhiteList {
+		k.log.Warn("tx from addr not in whitelist to create blockchain", "disableCreateChainWhiteList", k.disableCreateChainWhiteList)
+		return ErrAddrNotInWhiteList
+	}
+	investment := desc.Tx.GetAmountByAddress(bcName)
+	k.log.Info("create blockchain", "chain", bcName, "investment", investment, "need", k.minNewChainAmount)
+	if investment.Cmp(k.minNewChainAmount) < 0 {
+		return ErrNoEnoughUTXO
+	}
+	err = k.CreateBlockChain(bcName, []byte(bcData))
+	if err == ErrBlockChainExist { //暂时忽略
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	if k.register != nil {
+		k.log.Info("register block chain", "name", bcName)
+		return k.register.RegisterBlockChain(bcName)
+	}
+	return nil
+}
+
+// runStopBlockChain 为客户端提供停用制定链服务，不删除该链目录下内容
+func (k *Kernel) runStopBlockChain(desc *contract.TxDesc) error {
+	if desc.Args["name"] == nil {
+		k.log.Warn("Chain name in tx for stopping a blockchain should be a non-null value.")
+		return ErrInvalidChainName
+	}
+	bcName, ok := desc.Args["name"].(string)
+	if !ok || bcName == "" {
+		k.log.Warn("Chain name in tx for stopping a blockchain should be a non-empty string.")
+		return ErrInvalidChainName
+	}
+	if bcName == "xuper" {
+		k.log.Warn("Oops, main-chain:xuper cannot be stopped.")
+		return ErrPermissionDenied
+	}
+	if !desc.Tx.FromAddrInList(k.newChainWhiteList) && !k.disableCreateChainWhiteList {
+		k.log.Warn("Address in tx for stopping a blockchain should be in whitelist", "disableCreateChainWhiteList", k.disableCreateChainWhiteList, "ChainWhiteList", k.newChainWhiteList)
+		return ErrAddrNotInWhiteList
+	}
+	if err := k.register.UnloadBlockChain(bcName); err != nil {
```

Comment:
If xchain restarted, it will be reloaded,whether the directory can be renamed?

Reply:
Moving deleted-chain dictionary is complicated when deal with rollback operation, which should be atomic, so we put it forward.

Reply:
> If xchain restarted, it will be reloaded,whether the directory can be renamed?

User need to move the chain to other folder after it has been stopped.